preprocessors/openstreetmap/osm_service.py

```diff
@@ -372,135 +326,97 @@ def compute_new_node(node_params, bearing, bbox_coordinates):
-        y2 = radians(lat_min)
-        x2 = ((math.tan(theta) * (y2 - lat1)) /
-              math.cos(lat1)) + lon1
-        y2 = degrees(y2)
-        x2 = degrees(x2)
-        # If true, the intercept is at the bottom side
-        if ((y2 >= lat_min and y2 <= lat_max) and
-                (x2 >= lon_min and x2 <= lon_max)):
-            lat2 = y2
-            lon2 = x2
-        else:  # Otherwise, the intercept is at the right side,
-            # so compute for latitude while set longitude to
-            # lon_max.
-            x2 = radians(lon_max)
-            y2 = ((x2 - lon1) * math.cos(lat1) / math.tan(theta))
-            lat2 = degrees(y2 + lat1)
-            lon2 = degrees(x2)
-    elif bearing > 180 and bearing < 270:  # The street intercepts
-        # either the bottom or the left side of the bounds.
-        y2 = radians(lat_min)
-        x2 = ((math.tan(theta) * (y2 - lat1)) / math.cos(lat1))
-        y2 = degrees(y2)
-        x2 = degrees(x2 + lon1)
-        # If true, the intercept is at the bottom side
-        if ((y2 >= lat_min and y2 <= lat_max) and
-                (x2 >= lon_min and x2 <= lon_max)):
-            lat2 = y2
-            lon2 = x2
-        else:  # Otherwise, the intercept is at the left side,
-            # so compute for latitude while set longitude to
-            # lon_min.
-            x2 = radians(lon_min)
-            y2 = ((x2 - lon1) * math.cos(lat1) / math.tan(theta))
-            lat2 = degrees(y2 + lat1)
-            lon2 = degrees(x2)
-    elif bearing > 270 and bearing < 360:  # The street intercepts
-        # either the top or the left side of the bounds.
-        y2 = radians(lat_max)
-        x2 = ((math.tan(theta) * (y2 - lat1)) / math.cos(lat1))
-        y2 = degrees(y2)
-        x2 = degrees(x2 + lon1)
-        # If true, the intercept is at the top side
-        if ((y2 >= lat_min and y2 <= lat_max) and
-                (x2 >= lon_min and x2 <= lon_max)):
-            lat2 = y2
-            lon2 = x2
-        else:  # Otherwise, the intercept is at the left side,
-            # so compute for latitude while set longitude to
-            # lon_min.
-            x2 = radians(lon_min)
-            y2 = ((x2 - lon1) * math.cos(lat1) / math.tan(theta))
-            lat2 = degrees(y2 + lat1)
-            lon2 = degrees(x2)
-    elif bearing >= 0 or bearing == 360:  # The street intercepts only
-        # at the top side of the bounds, so compute for longitude while
-        # setting the latitude to lat_max.
-        y2 = radians(lat_max)
-        x2 = ((math.tan(theta) * (y2 - lat1)) / math.cos(lat1))
-        y2 = degrees(y2)
-        x2 = degrees(x2 + lon1)
-        lat2 = y2
-        lon2 = x2
-    elif bearing == 45:  # The street intercepts only
-        # at the NE edge of the bounds.
+    # The bearing indicates the side of the bounding box that intercepts the
+    # street.
+    flag = False
+    if bearing > 0 and bearing < 90:  # if true,  the intercept is at
```

Comment:
This is still a very long sequence of checks here that is difficult to read.

Reply:
That's true. The redundant checks have been removed.